librosa/filters.py

```diff
@@ -599,7 +599,7 @@ def constant_q(sr, fmin=None, n_bins=84, bins_per_octave=12, tuning=0.0,
     if pad_fft:
         max_len = int(2.0**(np.ceil(np.log2(max_len))))
     else:
-        max_len = np.ceil(max_len)
+        max_len = int(np.ceil(max_len))
```

Comment:
I believe this PR is a minimum fix for the test failure, but yes, it might be better implemented in `util.pad_center`, though it might affect functions that use `util.pad_center`.

Originally I thought two ways to fix:

1. Casting `size` (which is the second argument in `util.pad_center`)  to be `int` type regardless of the type of `size` (i.e., `size = int(size)`)
2. Throwing an error if `size` is not a integral type (that current implementation does)

My decision was the second one: throwing `TypeError` when float-valued `size` is supplied, since the first one might causes unexpected results if it doesn't assume the `size` to be only int-valued.

I guess `size` is assumed to be integer type, but I din't realized that what exactly it does. Can I ask you what do you think how to fix this in `util.pad_center`? Thank you for your feedback!